src/core_init_atmosphere/mpas_geotile_manager.F

```diff
@@ -0,0 +1,1025 @@
+        real (kind=RKIND), pointer :: dy
+        integer, pointer :: tile_nx
+        integer, pointer :: tile_ny
+        integer, dimension(2) :: x
+        integer, dimension(2) :: y
+
+        integer :: ierr
+        ierr = 0
+
+        call mpas_pool_get_config(mgr % pool, 'tile_x', tile_nx)
+        call mpas_pool_get_config(mgr % pool, 'tile_y', tile_ny)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        ! Find the global pixel location that contains lat, lon
+        call mgr % latlon_to_pixel(lat, lon, x(1), y(1))
+
+        ! Calculate the range of this tile, which will be its filename
+        x(1) = (x(1) - modulo(x(1), tile_nx)) + 1
+        x(2) = x(1) + tile_nx
+        x(2) = x(2) - modulo(x(2), tile_nx)
+
+        y(1) = (y(1) - modulo(y(1), tile_ny)) + 1
+        y(2) = y(1) + tile_ny
+        y(2) = y(2) - modulo(y(2), tile_ny)
+
+        write(fname, fname_format) x(1), x(2), y(1), y(2)
+
+        if (present(start_x)) then
+            start_x = x(1)
+        endif
+        if (present(start_y)) then
+            start_y = y(1)
+        endif
+
+    end function mpas_geotile_mgr_gen_tile_name
+
+
+    !***********************************************************************
+    !
+    !  public subroutine mpas_geotile_mgr_tile_to_latlon => tile_to_latlon
+    !
+    !> \brief   Translate a tile indices to latitude and longitude,
+    !> \author  Miles A. Curry
+    !> \date    02/20/2020
+    !> \details
+    !> Given a tile, translate the relative tile coordinates, i, j, of that
+    !> tile to a latitude and longitude coordinate. Upon success, lat, lon
+    !> will be in the range of -1/2 * pi to 1/2 * pi and 0 to 2.0 * pi, respectively.
+    !
+    !-----------------------------------------------------------------------
+    subroutine mpas_geotile_mgr_tile_to_latlon(mgr, tile, j, i, lat, lon)
+
+        implicit none
+
+        class (mpas_geotile_mgr_type) :: mgr
+        type (mpas_geotile_type), pointer, intent(in) :: tile
+        integer, value :: j
+        integer, value :: i
+        real (kind=RKIND), intent(out) :: lat
+        real (kind=RKIND), intent(out) :: lon
+
+        integer, pointer :: tile_bdr
+        real (kind=RKIND), pointer :: known_lon
+        real (kind=RKIND), pointer :: known_lat
+        real (kind=RKIND), pointer :: dx
+        real (kind=RKIND), pointer :: dy
+        integer :: ierr
+
+        ierr = 0
+
+        call mpas_pool_get_config(mgr % pool, 'tile_bdr', tile_bdr)
+        call mpas_pool_get_config(mgr % pool, 'known_lat', known_lat)
+        call mpas_pool_get_config(mgr % pool, 'known_lon', known_lon)
+        call mpas_pool_get_config(mgr % pool, 'dx', dx)
+        call mpas_pool_get_config(mgr % pool, 'dy', dy)
+
+        lat = real((j - (tile_bdr + 1) + tile % y - 1), kind=RKIND) * dy + known_lat + 1.0
+        lon = real((i - (tile_bdr + 1) + tile % x - 1), kind=RKIND) * dx + known_lon + 1.0
```

Comment:
Where does the `+ 1.0` at the end of the expressions for `lat` and `lon` come from?

Reply:
I'm not sure where it comes from. Previous implementations didn't have it. On its own, it doesn't make much sense to add 1.0 to the latitude or longitude.